Gems/Archive/Code/Source/Tools/ArchiveWriter.cpp

```diff
@@ -576,27 +588,37 @@ namespace Archive
             return result;
         }
 
-        // Initialize the tocWriteSpan to the raw Toc Data
+        // Initialize the tocWriteSpan to the tocRawBuffer above
         AZStd::span<const AZStd::byte> tocWriteSpan = rawTocResult.m_tocSpan;
 
         // Check if the table of contents should be compressed
-        if (m_archiveHeader.m_tocCompressionAlgoIndex < UncompressedAlgorithmIndex)
+        Compression::CompressionAlgorithmId tocCompressionAlgorithmId =
+            m_archiveHeader.m_tocCompressionAlgoIndex < UncompressedAlgorithmIndex
+            ? m_archiveHeader.m_compressionAlgorithmsIds[m_archiveHeader.m_tocCompressionAlgoIndex]
+            : Compression::Uncompressed;
+        if (m_settings.m_tocCompressionAlgorithm.has_value())
         {
-            // Resize the compress block buffer to be the same size as the input buffer
-            AZStd::vector<AZStd::byte> tocCompressBuffer;
+            tocCompressionAlgorithmId = m_settings.m_tocCompressionAlgorithm.value();
```

Comment:
m_settings can override the toc compression algo id from the m_archiveheader? is it possible for the archive header to specify a different toc compression algo than m_settings?

Reply:
Yes, the Compression Algorithm is set by the previous instance of the `ArchiveWriterSettings` that was used to write the Archive before.
So if the a previous use of the `ArchiveWriterSettings` set the table of contents(TOC) compression algorithm to Zlib, but then the next use of the `ArchiveWriterSettings` set the TOC compression algorithm to LZ4, then the compression algorithm will be different than the one in the set in the archiveHeader